File: 2021/template.kt
import java.io.File

typealias TInput_DAY_ = List<Int>
typealias TOutput_DAY_ = Int

private fun part1(input: TInput_DAY_): TOutput_DAY_ = 0

private fun part2(input: TInput_DAY_): TOutput_DAY_ = 1

private fun parse(raw_input: String): TInput_DAY_ = nums(raw_input)

fun main() {
    val rawInput = File("_DAY_.in").readText()
    val input = parse(rawInput)

    println("Solution for Part 1: ${part1(input)}")
    println("Solution for Part 2: ${part2(input)}")
}

// Solution part 1:
// Solution part 2:

File: 2022/template.hs
#! /usr/bin/env -S runhaskell
import Data.List
import Data.List.Split
import Data.Char

import Util

type Input = String
type Output = Int

parse :: String -> Maybe Input
parse = Just

part1 :: Input -> Output
part1 _ = 0

part2 :: Input -> Output
part2 _ = 1

main :: IO()
main = do
    f <- readFile("_DAY_.in")
    let input = parse f
    case input of
      Nothing -> putStrLn "Unable to read input."
      Just i -> do
        putStr "Solution for part 1: "
        putStrLn . show $ part1 i
        putStr "Solution for part 2: "
        putStrLn . show $ part2 i

-- Solution part 1:
-- Solution part 2:
